src/components/tables/TableConnected.tsx

```diff
@@ -0,0 +1,113 @@
+import { ITableState, ITableCompositeState } from './TableReducers';
+import { addActionsToActionBar } from '../actions/ActionBarActions';
+import { addTable, removeTable } from './TableActions';
+import { IReactVaporState } from '../../ReactVapor';
+import { ReduxUtils } from '../../utils/ReduxUtils';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import * as _ from 'underscore';
+import { IActionOptions } from '../actions/Action';
+import { defaultTableStateModifierThunk } from './TableDataModifier';
+import { IDropdownOption } from '../dropdownSearch/DropdownSearch';
+import { selectOptionDropdownSearch, closeDropdownSearch } from '../dropdownSearch/DropdownSearchActions';
+import { IPerPageState } from '../navigation/perPage/NavigationPerPageReducers';
+import { IPaginationState } from '../navigation/pagination/NavigationPaginationReducers';
+import { IFilterState } from '../filterBox/FilterBoxReducers';
+import { ITableHeaderCellState } from './TableHeaderCellReducers';
+import { IDropdownSearchState } from '../dropdownSearch/DropdownSearchReducers';
+import { contains } from 'underscore.string';
+import { IDispatch } from '../../utils/ReduxUtils';
+
+export type IAttributeValue = any;
+export interface IPredicateAttributes {
+  [attributeName: string]: IAttributeValue;
+};
+
+export interface ITableDispatchProps {
+  onDidMount: () => void;
+  onUnmount: () => void;
+  onModifyData: (
+    shouldResetPage: boolean,
+    tableCompositeState: ITableCompositeState,
+    previousTableCompositeState?: ITableCompositeState,
+  ) => void;
+  onRowClick: (actions: IActionOptions[]) => void;
+  onPredicateOptionClick: (predicateId: string, option: IDropdownOption) => void;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: ITableOwnProps) => {
+  const tableState: ITableState = state.tables[ownProps.id] || {} as ITableState;
+  const filterState: IFilterState = tableState && _.findWhere(state.filters, { id: tableState.filterId });
+  const paginationState: IPaginationState = tableState && _.findWhere(state.paginationComposite, { id: tableState.paginationId });
+  const perPageState: IPerPageState = tableState && _.findWhere(state.perPageComposite, { id: tableState.perPageId });
+  const tableHeaderCellState: ITableHeaderCellState = tableState && state.tableHeaderCells[tableState.tableHeaderCellId];
+  const predicateStates: IDropdownSearchState[] = tableState && _.reject(state.dropdownSearch, (dropdownSearch: IDropdownSearchState) => !contains(dropdownSearch.id, ownProps.id)) || [];
+
+  return {
+    tableCompositeState: {
+      id: tableState.id,
+      data: tableState.data,
+      isInError: tableState.isInError,
+      isLoading: tableState.isLoading,
+      filter: filterState && filterState.filterText || '',
+      page: paginationState && paginationState.pageNb || 0,
+      perPage: perPageState && perPageState.perPage || DEFAULT_TABLE_PER_PAGE,
```

Comment:
`DEFAULT_TABLE_PER_PAGE` is already in Table's default props. You can remove it from here. `0` and `''` should also be default props.

Reply:
moved in default props, does not affect UTs